crates/turborepo-micro-frontend/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+mod error;
+
+use std::collections::BTreeMap;
+
+use biome_deserialize_macros::Deserializable;
+use biome_json_parser::JsonParserOptions;
+pub use error::Error;
+use serde::Serialize;
+use turbopath::AbsoluteSystemPath;
+
+pub const MICRO_FRONTENDS_CONFIG: &str = "micro-frontends.jsonc";
+pub const MICRO_FRONTENDS_PACKAGES: &[&str] = ["@vercel/micro-frontends-internal"].as_slice();
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Config {
+    pub version: String,
+    #[serde(rename = "$schema")]
+    pub schema: Option<String>,
+    pub applications: BTreeMap<String, Application>,
+    pub options: Option<Options>,
+}
+
+impl Config {
+    /// Reads config from given path.
+    /// Returns `Ok(None)` if the file does not exist
+    pub fn load(config_path: &AbsoluteSystemPath) -> Result<Option<Self>, Error> {
+        let Some(contents) = config_path.read_existing_to_string()? else {
+            return Ok(None);
+        };
+        let config = Self::from_str(&contents, config_path.as_str()).map_err(Error::biome_error)?;
+        Ok(Some(config))
+    }
+
+    pub fn from_str(input: &str, source: &str) -> Result<Self, Vec<biome_diagnostics::Error>> {
+        let (config, errs) = biome_deserialize::json::deserialize_from_json_str(
+            input,
+            JsonParserOptions::default().with_allow_comments(),
+            source,
+        )
+        .consume();
+        if let Some(config) = config {
+            Ok(config)
+        } else {
+            Err(errs)
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Options {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub vercel: Option<VercelOptions>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct VercelOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stitch_applications_in_preview: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bypass_deployment_protection_in_production: Option<bool>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+pub struct Application {
+    // default = true -> no routing
+    // default = false -> requires routing
+    pub default: bool,
+    pub routing: Option<ZoneRouting>,
+    pub development: Development,
+    pub production: Host,
+    pub metadata: Option<BTreeMap<String, String>>,
+    pub federation: Option<Federation>,
+    pub vercel: Option<Vercel>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserializable, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Vercel {
```

Comment:
this is missing `default_route`

Reply:
I might remove a majority of these fields and only keep the ones that we need (which apps are part of the MFE and what they're development tasks are). This was written when the local proxy was going to be in Rust and this information would have been useful.